drivers/overlay/overlay.go

```diff
@@ -162,3 +150,51 @@ func (d *driver) configure() error {
 func (d *driver) Type() string {
 	return networkType
 }
+
+func (d *driver) nodeJoin(node net.IP, self bool) {
+	d.Lock()
+	if self {
+		d.bindAddress = node.String()
+	} else {
+		d.neighIP = node.String()
```

Comment:
Should we avoid to set  `neighIP` with the new node discovery updates coming set if already set ?
I know you wouldn't join again because of the sync.Once{}, but it seems the neighbor IP would be overwritten with the latest discovered node IP.

if what I described above corresponds to the current behavior, then it means we do not need to store the neighbor IP because we allow it to be overwritten, a function variable would do it and you would pass the variable to `serfJoin()`.